src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorTask.java

```diff
@@ -272,12 +285,15 @@ Offsets<YBPartition, YugabyteDBOffsetContext> getPreviousOffsetsFromProviderAndL
             Offsets.of(reader.offsets(partitions));
 
         boolean found = false;
-        for (YBPartition partition : partitions) {
-            YugabyteDBOffsetContext offset = offsets.getOffsets().get(partition); //offsets.get(partition);
-
-            if (offset != null) {
-                found = true;
-                LOGGER.info("Found previous partition offset {}: {}", partition, offset);
+        if (offsets != null) {
+            found = true;
+
```

Comment:
Making the log changes started giving OOM issues, it could be related so I am planning to introduce those changes separately in another PR later so that we keep things isolated.